src/test/unit/micm/micm_wrapper.cpp

```diff
@@ -5,75 +5,56 @@
+  bool something_changed = false;
+  for(int i = 0; i < initial_concnetrations.size(); ++i)
+  {
+    if (state.GetOrderedConcentrations()[i] != initial_concnetrations[i])
+    {
+      something_changed = true;
+      break;
+    }
+  }
+  EXPECT_TRUE(something_changed);
 }
 
-TEST(MICMWrapper, CanParseCBVV0)
+TEST(MICMWrapper, Rosenbrock)
 {
-  musica::MICM micm;
-  musica::Chemistry chemistry = musica::ReadConfiguration("configs/carbon_bond_5");
-  EXPECT_EQ(chemistry.system.gas_phase_.species_.size(), 67);
-  EXPECT_EQ(chemistry.processes.size(), 200);
+  DoChemistry(musica::MICMSolver::Rosenbrock);
 }
 
-TEST(MICMWrapper, CanParseTS1V0)
+TEST(MICMWrapper, RosenbrockStandardOrder)
 {
-  musica::MICM micm;
-  musica::Chemistry chemistry = musica::ReadConfiguration("configs/TS1");
-  EXPECT_EQ(chemistry.system.gas_phase_.species_.size(), 210);
-  EXPECT_EQ(chemistry.processes.size(), 547);
+  DoChemistry(musica::MICMSolver::RosenbrockStandardOrder);
 }
 
-TEST(MICMWrapper, DetectsInvalidConfigV0)
+TEST(MICMWrapper, BackwardEuler)
 {
-  musica::MICM micm;
-  EXPECT_ANY_THROW(musica::ReadConfiguration("configs/invalid"));
+  DoChemistry(musica::MICMSolver::BackwardEuler);
 }
 
-TEST(MICMWrapper, CanParseChapmanV1)
+TEST(MICMWrapper, BackwardEulerStandardOrder)
 {
-  std::vector<std::string> extensions = { ".json", ".yaml" };
-
-  for (const auto& extension : extensions)
-  {
-    musica::MICM micm;
-    musica::Chemistry chemistry = musica::ReadConfiguration("configs/v1/chapman/config" + extension);
-    EXPECT_EQ(chemistry.system.gas_phase_.species_.size(), 5);
-    EXPECT_EQ(chemistry.processes.size(), 7);
-    EXPECT_EQ(chemistry.system.phases_.size(), 0);
-    EXPECT_EQ(chemistry.system.gas_phase_.species_[0].name_, "M");
-    EXPECT_NE(chemistry.system.gas_phase_.species_[0].parameterize_, nullptr);
-    EXPECT_EQ(chemistry.system.gas_phase_.species_[1].name_, "O");
-    EXPECT_EQ(chemistry.system.gas_phase_.species_[2].name_, "O2");
-    EXPECT_EQ(chemistry.system.gas_phase_.species_[3].name_, "O3");
-    EXPECT_EQ(chemistry.system.gas_phase_.species_[4].name_, "O1D");
-
-    EXPECT_EQ(chemistry.system.gas_phase_.species_[3].GetProperty<std::string>("__long name"), "ozone");
-  }
+  DoChemistry(musica::MICMSolver::BackwardEulerStandardOrder);
 }
 
-TEST(MICMWrapper, CanParseFullV1)
+#ifdef MUSICA_ENABLE_CUDA
+TEST(MICMWrapper, CudaRosenbrock)
 {
-  std::vector<std::string> extensions = { ".json", ".yaml" };
-
-  for (const auto& extension : extensions)
-  {
-    musica::MICM micm;
-    musica::Chemistry chemistry = musica::ReadConfiguration("configs/v1/full_configuration" + extension);
-    EXPECT_EQ(chemistry.system.gas_phase_.species_.size(), 4);
-    EXPECT_EQ(chemistry.system.gas_phase_.name_, "gas");
-    EXPECT_EQ(chemistry.system.phases_.size(), 3);
-    EXPECT_EQ(chemistry.processes.size(), 9);
-  }
-}
\ No newline at end of file
+  DoChemistry(musica::MICMSolver::CudaRosenbrock);
+}
+#endif
```

Comment:
similar comment above

Reply:
Again, need something in the code to let us know cuda is being used